core/src/test/java/io/questdb/cairo/CreateTableTest.java

```diff
@@ -161,6 +161,197 @@ public void testCreateTableAsSelectInheritsColumnIndex() throws Exception {
+        assertQuery("s1\ts2\ts3\n", "select * from x", "create table x(like tab)", null);
+        assertColumnsIndexed("x", "s1", "s2", "s3");
+    }
+
+    @Test
+    public void testCreateTableFromLikeTableWithPartition() throws Exception {
+        assertCompile("create table x (" +
+                "a INT," +
+                "t timestamp) timestamp(t) partition by MONTH");
+        assertQuery("a\tt\n", "select * from tab", "create table tab (like x)", "t");
+        assertPartitionIndex("tab", PartitionBy.MONTH, 1);
+    }
+
+    @Test
+    public void testCreateTableLikeTableWithSymbolCapacity() throws Exception {
+        int symbolCapacity = 128;
+
+        assertCompile("create table x (" +
+                "a INT," +
+                "y SYMBOL CAPACITY " + symbolCapacity + " NOCACHE," +
+                "t timestamp) timestamp(t) partition by MONTH");
+        assertQuery("a\ty\tt\n", "select * from tab", "create table tab ( like x)", "t");
+        SymbolParameters parameters = new SymbolParameters(symbolCapacity, false, false, null);
+        assertSymbolParameters("tab", 1, parameters);
+    }
+
+    @Test
+    public void testCreateTableLikeTableWithCachedSymbol() throws Exception {
+        boolean isSymbolCached = true;
+        String symbolCacheParameterValue = isSymbolCached ? "CACHE" : "NOCACHE";
+
+        assertCompile("create table x (" +
+                "a INT," +
+                "y SYMBOL " + symbolCacheParameterValue + "," +
+                "t timestamp) timestamp(t) partition by MONTH");
+        assertQuery("a\ty\tt\n", "select * from tab", "create table tab ( like x)", "t");
+        SymbolParameters parameters = new SymbolParameters(null, isSymbolCached, false, null);
+        assertSymbolParameters("tab", 1, parameters);
+    }
+
+    @Test
+    public void testCreateTableLikeTableWithIndexBlockCapacity() throws Exception {
+        int indexBlockCapacity = 128;
+        assertCompile("create table x (" +
+                "a INT," +
+                "y SYMBOL NOCACHE INDEX CAPACITY " + indexBlockCapacity + "," +
+                "t timestamp) timestamp(t) partition by MONTH");
+        assertQuery("a\ty\tt\n", "select * from tab", "create table tab ( like x)", "t");
+        SymbolParameters parameters = new SymbolParameters(null, false, true, indexBlockCapacity);
+        assertSymbolParameters("tab", 1, parameters);
+    }
+
+    @Test
+    public void testCreateTableIfNotExistsLikeTable() throws Exception {
+        assertCompile("create table x (s1 symbol) ");
+        assertCompile("create table y (s2 symbol)");
+        assertQuery("s1\n", "select * from x", "create table if not exists x (like y)", null);
+    }
+
+    @Test
+    public void testCreateTableLikeTableWithWALEnabled() throws Exception {
+        boolean isWalEnabled = true;
+        String walParameterValue = isWalEnabled ? "WAL" : "NONE";
```

Comment:
Does `NONE` value work? From what I see in `SqlParser`, to disable WAL you have to use either an empty string or `BYPASS WAL` string.

Reply:
Done